### ISSUE TITLE
Wishlist: provide a way to process an array

### ISSUE THREAD

**Issue by chocolateboy** [ivanwills/node-getopt-long#4]
Getopt::Long has [`GetOptionsFromArray`](http://search.cpan.org/perldoc?Getopt%3A%3ALong#Parsing_options_from_an_arbitrary_array), but there doesn't appear to be a way to do this in node-getopt-long. Among other things, it would be useful for [testing](https://github.com/ivanwills/node-getopt-long/issues/3).

Overloading `process` seems the most obvious way to implement this. It could take an optional array and default to `process.argv` if it's not supplied:

```javascript
var getopt = Getopt.configure(...):

var optionsFromArray = getopt.process([ '--foo', 'bar', '--baz' ]);
var optionsFromArgv  = getopt.process();
```

**Comment by ivanwills** [comment]
That's easy enough to do but can I ask what's the situation you are trying to achieve?

**Comment by ivanwills** [comment]
Fixed in v0.3.1
